andi/andi.py

```diff
@@ -342,14 +346,23 @@ def _plan(class_or_func: Callable, *,
         if sel_cls is not None:
             errors = []  # type: List[Tuple]
             if sel_cls not in plan_od:
-                plan, errors = _plan(sel_cls,
-                                     is_injectable=is_injectable,
-                                     externally_provided=externally_provided,
-                                     full_final_kwargs=True,
-                                     dependency_stack=dependency_stack,
-                                     overrides=arg_overrides,
-                                     recursive_overrides=recursive_overrides)
-                plan_od.update(plan)
+                skip = False
+                if is_typing_annotated(sel_cls):
+                    sel_cls_stripped = strip_annotated(sel_cls)
+                    if sel_cls_stripped in seen_annotated:
+                        # warning/error/skip/not skip?
```

Comment:
Would it make sense to preserve duplicates instead, as soon as they have different annotations? Why are they skipped?

I'm also not sure I understand the logic. It seems only annotated dependencies are considered for duplicate checks, but there could also be non-annotated dependencies of the same type - is this case handled?

Reply:
> Would it make sense to preserve duplicates instead, as soon as they have different annotations?

I think we discussed this and decided that for now we should omit duplicates and log a warning. 

> there could also be non-annotated dependencies of the same type - is this case handled?

If one is annotated but another isn't? Yeah, that's not currently considered which is an oversight.

Reply:
> I think we discussed this and decided that for now we should omit duplicates and log a warning.

I was thinking it was because of the limitation of the approach where the annotations are returned separately; it was the only way in that case. 

Here it's less clear that we need to do this at andi level. Maybe doing it on scrapy-poet level is better, as sometimes the duplicates are allowed, and sometimes they are not, and it looks like this is a provider's decision.

Reply:
Although it may need more though, or maybe even a parameter to control this.
At the same time, it seems ay default behavior is ok for now (allow duplicates, disallow them), it seems it shouldn't block us.

Reply:
Now no deduplication happens.

Reply:
Does the deduplication happen if the annotated class and parameters are the same?